### PR TITLE
Upgrade FB SDK and fixes some compilation issues on Xcode 8

### PR DESCRIPTION
**Pull request by migue1s**
my Habitica User-ID: 602caa83-e061-42b3-a715-787887e8050d

This pull request fixes some project configurations. The project settings didn't allow it to be compiled correctly in Xcode 8 due to some outdated swift code inside the Facebook sdk. 
Additionally, it also upgrades the Gemfile as the Podfile.lock was created with cocoa pods 1.2 but it only specified 1.0> as the requirement, up until cocoa pods 1.1 there were many problems with Xcode 8.

Detailed changes:
- Changes the cocoa pods version to 1.2 in the gem file (the Podfile.lock was already using 1.2 but the gem file specified >1.0)
- Upgrades restkit to 0.27 (to make it compatible with cocoa pods 1.2)
- Replaces deprecated FB SDK for the new one
- Links the FB login button delegate in the storyboard (it was somehow being set magically?)

### CODE REVIEW

File: Podfile [hunk 1]
```diff
@@ -19,7 +19,7 @@ plugin 'cocoapods-keys', {
 target 'Habitica' do
 pod 'CRToast', :git => 'https://github.com/cruffenach/CRToast'
 pod 'FontAwesomeIconFactory'
-pod 'RestKit', :git => 'https://github.com/RestKit/RestKit.git', :commit => '13d98d5a6a5e06656ad040013dcae149b7cf8b99'
+pod 'RestKit'
```
Comment (vIiRuS) [inline Podfile]: The RestKit commit referenced in the podfile is actually newer than the 0.27 release (and is still the most recent commit in the development branch) and iirc it fixed issues that happened when the project was built for iOS 10. Could you revert that to the specific commit?

**Comment by migue1s** [comment]
I see, my bad! Thought the commit was before since I had trouble compiling restkit in the start 😅 Just reverted back to that version + added the correct padding to the podfile.

File: Podfile (hunk 1 above)
Comment (migue1s) [inline Podfile]: Got it, just reverted back to what it was before.

**Comment by vIiRuS** [comment]
Ok!

I noted this as a contribution in your habitica profile

**Pull request merged**